### PR TITLE
Enforce single canvas view for annotation editor

### PR DESCRIPTION
**Pull request by lutzhelm**
As there are currently some problems with annotation creation / editing in views with more than one visible canvas (see #24), this PR adds a dialog that is displayed if a user clicks the "New Annotation" icon in any other than `single` view.

What's still missing in this PR is a similar behavior for the edit button.

### CODE REVIEW

**Comment by lutzhelm** [comment]
At 901709d08fba1deffb1f9663424f95128a641bcc, switching to the single canvas view also opened the companion window. But I think this would introduce to much complexity especially for editing existing annotations.

File: src/plugins/canvasAnnotationsPlugin.js [hunk 1]
```diff
@@ -26,12 +46,21 @@ class CanvasAnnotationsWrapper extends Component {
...
           storageAdapter: config.annotation.adapter,
+          toggleSingleCanvasDialogOpen: this.toggleSingleCanvasDialogOpen,
           windowId: targetProps.windowId,
+          windowViewType,
         }}
       >
         <TargetComponent
           {...props} // eslint-disable-line react/jsx-props-no-spreading
         />
+        {windowViewType !== 'single' && (
```
Comment (mejackreed) [inline src/plugins/canvasAnnotationsPlugin.js:57]: Can you say more about why we need to render the component here as well? I'm not sure of the case when this should render. Thanks!

Comment (mejackreed) [reply src/plugins/canvasAnnotationsPlugin.js:57]: Ah.. nevermind. ++ @camillevilla for pointing me to the editing view.

**Review by mejackreed — COMMENTED** [review]
Overall this looks really good. One quick question I have is if we maybe want to start looking at expanding (or moving) what is done in `AnnotationCreation` to handle the logic here rather than on our button clicks, or if there are reasons that wasn't done.

**Comment by lutzhelm** [comment]
> Overall this looks really good. One quick question I have is if we maybe want to start looking at expanding (or moving) what is done in `AnnotationCreation` to handle the logic here rather than on our button clicks, or if there are reasons that wasn't done.

I'm not entirely sure I understand the question correctly. There is some multi canvas logic in `AnnotationCreation` that is, at least for the moment, no longer relevant as soon as this PR is merged. However annotations across multiple canvases might still be a valid use case. In the end, fixing #24 and removing this workaround should be the goal.

**Comment by lutzhelm** [comment]
I'll look into the fresh conflicts.

**Pull request merged**